### PR TITLE
Represent service waypoint ingress binding in service condition

### PR DESCRIPTION
**Pull request by tjons**
**Please provide a description of this PR:**

This PR resolves https://github.com/istio/istio/issues/53845 by adding new messages to represent a service binding to a waypoint proxy for ingress.

### CODE REVIEW

File: pilot/pkg/model/service.go [hunk 1]
```diff
@@ -1035,14 +1041,36 @@ func (i ServiceInfo) GetConditions() ConditionSet {
...
+
+	if i.Waypoint.IngressUseWaypoint {
+		set[IngressUsingWaypoint] = []Condition{
+			{
+				Status:  true,
+				Reason:  string(WaypointAccepted),
+				Message: fmt.Sprintf("Successfully attached to waypoint %v. Ingress traffic will traverse the waypoint", i.Waypoint.ResourceName),
+			},
+		}
+	} else if writeError && i.Waypoint.Error != nil {
```
Comment (ilrudie) [inline pilot/pkg/model/service.go:1064]: do we actually want to write a full second "attached to waypoint" condition with ingress note added on? If we want a second condition maybe we don't repeat the verbiage of the waypoint bound condition and just say, "Ingress traffic will traverse waypoint <name>". I think we could just have 1 condition and append the note about ingress use if it's applicable.

**Comment by howardjohn** [comment]
My main concern from https://github.com/istio/istio/issues/53845#issuecomment-2471226491 is that this is just 1 annotation and a relatively niche one. Istio has >50 annotations, not to mention other config sources. We cannot add conditions for each. I would prefer to just have a more aggregated way to report that things are not quite right

**Comment by tjons** [comment]
@howardjohn totally understand... I opened this to share my thoughts and get the conversation flowing again.

For representing this as a single condition, I'd propose something like this when the `istio.io/ingress-use-waypoint` label value is malformed:
```
  - lastTransitionTime: "2024-11-26T03:41:55Z"
    message: Successfully attached to waypoint default/waypoint. Ingress traffic is not using the waypoint, set the istio.io/ingress-use-waypoint label to true if desired.
    reason: WaypointAccepted
    status: "True"
    type: istio.io/WaypointBound
```

Or possibly just:
```
  - lastTransitionTime: "2024-11-26T03:41:55Z"
    message: Successfully attached to waypoint default/waypoint. Ingress traffic is not using the waypoint.
    reason: WaypointAccepted
    status: "True"
    type: istio.io/WaypointBound
```

let me know how you'd like to proceed and I'll make the change cc @ilrudie

**Comment by howardjohn** [comment]
That looks great to me. Thanks!

**Comment by ilrudie** [comment]
+1, looks good. Resolves my comment about the redundant verbiage in the second condition as well.

**Review by howardjohn — APPROVED** [review]
thanks!

**Pull request merged**